### DISCUSSION TITLE
[Question] How to set the gravity mode of object?

### DISCUSSION THREAD

**Discussion by tesollo-admin** [isaac-sim/IsaacLab#529] [Q&A]
Hi,

When setting up my scene in isaac lab, if I defined an object using `RigidObjectCfg` and initially set `disable_gravity=False` via `rigid_pros` parameter using `sim_utils.RigidBodyPropertiesCfg`, how can I change it to `disable_gravity=True (means zero gravity mode)` at a specific point during the simulation?

If it doesn't work, Is there any solution?

Thank you in advance.

**Comment by Mayankm96** [comment]
Hi @cjoattesollo ,

If you go to the [root PhysX view](https://isaac-sim.github.io/IsaacLab/source/api/lab/omni.isaac.lab.assets.html#omni.isaac.lab.assets.RigidObject.root_physx_view) for the rigid body asset class, there is a function named: `set_disable_gravities` which allows you to set this attribute.

You can probably do something like this (not tested):

```python
current_gravity_status = my_cube.root_physx_view.get_disable_gravities()
current_gravity_status[env_ids] = 0  # need to check if 0 means enable or disable
my_cube.root_physx_view.get_disable_gravities(current_gravity_status, env_ids)
```

**Reply by tesollo-admin** [reply]
Thank you for the quick reply. I will try and get back with the result!

**Reply by tesollo-admin** [reply]
It works well with `object.root_physx_view.set_disable_gravities(current_gravity_status, env_ids(Tensor))` . Thank you again :)

Checkpoint: `0: disable_gravity=False(with gravity) / 1: disable_gravity=True(without gravity) `

**Comment by strn18** [comment]
Does anyone know why this doesn't work for me?
I'm trying to make an object start without gravity and then enable gravity for it after one second.
Here’s my scene configuration:

```py
@configclass
class MySceneCfg(InteractiveSceneCfg):
    # objects
    object: RigidObjectCfg = RigidObjectCfg(
        prim_path="{ENV_REGEX_NS}/object",
        spawn=sim_utils.UsdFileCfg(
            usd_path=".../objec.usd",
            rigid_props=sim_utils.RigidBodyPropertiesCfg(
                kinematic_enabled=False,
                disable_gravity=True,
                enable_gyroscopic_forces=True,
                solver_position_iteration_count=8,
                solver_velocity_iteration_count=0,
                sleep_threshold=0.005,
                stabilization_threshold=0.0025,
                max_depenetration_velocity=1000.0,
            ),
            mass_props=sim_utils.MassPropertiesCfg(density=400.0),
            visual_material=sim_utils.PreviewSurfaceCfg(diffuse_color=(0.95, 0.8, 0.3)),
        ),
        init_state=RigidObjectCfg.InitialStateCfg(pos=(0.0, 0.0, 0.5), rot=(1.0, 0.0, 0.0, 0.0)),
    )

    # lights
    light = AssetBaseCfg(
        prim_path="/World/light",
        spawn=sim_utils.DistantLightCfg(color=(0.95, 0.95, 0.95), intensity=1000.0),
    )

    dome_light = AssetBaseCfg(
        prim_path="/World/domeLight",
        spawn=sim_utils.DomeLightCfg(color=(0.02, 0.02, 0.02), intensity=1000.0),
    )
```

As shown, I set `disable_gravity=True` in the object’s rigid body properties so that it stays fixed at its initial position when the environment starts.

After one second, I run the following function (triggered by one of my `EventTerm`):

```py
def enable_object_gravity(
    env: ManagerBasedEnv,
    env_ids: torch.Tensor,
    asset_cfg: SceneEntityCfg = SceneEntityCfg("object"),
):
    """Enable gravity for the given asset.
    """
    # extract the used quantities (to enable type-hinting)
    asset: RigidObject = env.scene[asset_cfg.name]

    cur_gravity_status = asset.root_physx_view.get_disable_gravities()
    cur_gravity_status[env_ids] = 0

    asset.root_physx_view.set_disable_gravities(cur_gravity_status, env_ids.to(cur_gravity_status.device))
```

I confirmed that:
- The function is indeed called one second after the environment starts. 
- The value of `asset.root_physx_view.get_disable_gravities()` changes to 0 after execution. 

However, despite these checks, the object does not fall as expected. It stays in its initial position as if gravity were still disabled.

**Reply by Dongx1aoYang** [reply]
Hi, I encountered the same issue. Have you found the solution?

**Reply by strn18** [reply]
Not yet.. So I'm just setting gravity for the entire scene instead (Using `SimulationView.set_gravity()` function).
